test/tokenizedSubaccounts/PPTSA/PPTSA_Validation.t.sol

```diff
@@ -0,0 +1,502 @@
+    tsa.signActionData(action, "");
+
+    tradeData.desiredAmount = 2.0e18;
+    action.module = IMatchingModule(address(10));
+    action.data = abi.encode(tradeData);
+    vm.expectRevert(PrincipalProtectedTSA.PPT_InvalidModule.selector);
+    tsa.signActionData(action, "");
+
+    action.module = tradeModule;
+    tradeData.asset = address(markets["weth"].option);
+    action.data = abi.encode(tradeData);
+    vm.expectRevert(PrincipalProtectedTSA.PPT_InvalidAsset.selector);
+    tsa.signActionData(action, "");
+
+    tradeData.asset = address(markets["weth"].base);
+    action.data = abi.encode(tradeData);
+    tsa.signActionData(action, "");
+
+    vm.warp(block.timestamp + 1 days);
+    vm.expectRevert(PrincipalProtectedTSA.PPT_InvalidActionExpiry.selector);
+    tsa.signActionData(action, "");
+  }
+
+  /////////////////
+  //     RFQ     //
+  /////////////////
+  function testVerifyRfqParams() public {
+    _depositToTSA(10e18);
+    _executeDeposit(10e18);
+    int amount = 1e18;
+    uint price = 1e18;
+    uint64 expiry = uint64(block.timestamp + 7 days);
+    uint strike = 800e18;
+    uint price2 = 4e18;
+    uint strike2 = 400e18;
+    vm.startPrank(signer);
+
+    (IRfqModule.RfqOrder memory makerOrder, IRfqModule.TakerOrder memory takerOrder) =
+      _setupRfq(amount, price, expiry, strike, price2, strike2, true);
+    IActionVerifier.Action memory action = _createRfqAction(takerOrder);
+
+    takerOrder.orderHash = "";
+    action.data = abi.encode(takerOrder);
+    vm.expectRevert(PrincipalProtectedTSA.PPT_TradeDataDoesNotMatchOrderHash.selector);
+    tsa.signActionData(action, abi.encode(makerOrder.trades));
+
+    makerOrder.trades[0].asset = address(markets["weth"].base);
+    makerOrder.trades[1].asset = address(markets["weth"].base);
+    takerOrder.orderHash = keccak256(abi.encode(makerOrder.trades));
+    action.data = abi.encode(takerOrder);
+    vm.expectRevert(PrincipalProtectedTSA.PPT_InvalidParams.selector);
+    tsa.signActionData(action, abi.encode(makerOrder.trades));
+
+    makerOrder.trades[0].asset = address(markets["weth"].option);
+    takerOrder.orderHash = keccak256(abi.encode(makerOrder.trades));
+    action.data = abi.encode(takerOrder);
+    vm.expectRevert(PrincipalProtectedTSA.PPT_InvalidParams.selector);
+    tsa.signActionData(action, abi.encode(makerOrder.trades));
+
+    makerOrder.trades[1].asset = address(markets["weth"].option);
+
+    (IRfqModule.RfqOrder memory smallerMakerOrder,) = _setupRfq(amount, price, expiry, strike, price2, strike2, true);
+    IRfqModule.TradeData[] memory smallerTrades = new IRfqModule.TradeData[](1);
+    smallerTrades[0] = makerOrder.trades[0];
+    smallerMakerOrder.trades = smallerTrades;
+    takerOrder.orderHash = keccak256(abi.encode(smallerTrades));
+    action.data = abi.encode(takerOrder);
+    vm.expectRevert(PrincipalProtectedTSA.PPT_InvalidParams.selector);
+    tsa.signActionData(action, abi.encode(smallerTrades));
+
+    // flip strikes
```

Comment:
need this to be more specific. Is this making sure that vault is selling if `isLongSpread` but it's selling? If so should actually write that out.

Reply:
Yeah I added these more for me, but now I should make them more for everyone else. Adding more detail